src/screens/AddNewItemScreen/index.tsx

```diff
@@ -13,32 +13,57 @@ import { style } from './style';
 import 'react-native-gesture-handler';
 import { createAlertMessageHelper } from '../../helpers/createAlertMessageHelper';
 import { AddNewItemScreeBackButton } from '../../components/AddNewItemScreenBackButton';
-import { useNavigation } from '@react-navigation/native';
-import { AddNewItemScreenNavigationProps } from './type';
+import { useNavigation, useRoute } from '@react-navigation/native';
+import {
+  AddNewItemScreenNavigationProps,
+  AddNewItemScreenRouteProps,
+} from './type';
+import { editItemSagaAction } from '../../redux/actions/todoSagaActions/editItemSagaAction';
+import { createCurrentDateHelper } from '../../helpers/createCurrentDateHelper';
+import { stopNotificationHelper } from '../../helpers/stopNotificationHelper';
 
 export const AddNewItemScreen: FC = () => {
   const navigation = useNavigation<AddNewItemScreenNavigationProps>();
+  const route = useRoute<AddNewItemScreenRouteProps>();
   const dispatch = useDispatch();
 
+  const isEdit = route.params.isEdit;
+  const editItem = route.params.editItem;
+  const initialState = editItem?.text ?? '';
+
+  useEffect(() => {
+    setDate(editItem?.notificationDate ?? new Date());
```

Comment:
why in useEffect? 

for text you don't use useEffect
`const [text, setText] = useState(initialState);`

Reply:
because you need to know draw the current time or the time when the notification of the task being changed is drawn in the date input